### PR TITLE
fix(license): validate license id and short name

### PR DESCRIPTION
**Pull request by maierthomas**
- add validate function to ThriftValidate.java
- verify license id / short name in addLicense and updateLicense

closes sw360/sw360portal#329

### CODE REVIEW

**Comment by maierthomas** [comment]
`assertId` method in `SW360Asserts.java` should validate the correct document id?
How do I validate the doc id exactly?

**Comment by mcjaeger** [comment]
please consider `1*(ALPHA / DIGIT / "-" / "." )`

from https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60

ID string is like "Apache-2.0" or "MIT" or "GPL-2.0-only"

**Comment by mcjaeger** [comment]
(from discussion, Alex proposed) ... to use a new "assertLicenseId" not just assertId

**Comment by maierthomas** [comment]
Should we include also `"+"` or is this character deprecated?
The validation of the value is only for a **new** license

**Comment by maierthomas** [comment]
(from discussion) the character `"+"` is allowed

**Comment by adrian-evo** [comment]
Tested and the license short name is validated fine.

File: libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java [hunk 1]
```diff
@@ -244,6 +245,28 @@ public static void validateAttachment(AttachmentContent attachment) throws SW360
         attachment.setType(TYPE_ATTACHMENT);
     }
 
+    public static void validateLicenseIdMatch(License license) throws SW360Exception {
+        if (license.isSetId() && license.isSetShortname()) {
+            String message = "license short name must be equal with license id";
```
Comment (alexbrdn) [inline libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java]: I believe "equal to" is the grammatically correct phrase

File: libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java [hunk 2]
```diff
@@ -244,6 +245,28 @@ public static void validateAttachment(AttachmentContent attachment) throws SW360
...
+            assertEquals(license.getId(), license.getShortname(), message);
+        }
+    }
+
+    public static void validateLicense(License license) throws SW360Exception {
+        final Pattern pattern = Pattern.compile("[A-Za-z0-9\\-.+]*");
```
Comment (alexbrdn) [inline libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java]: please extract the regexp into a named constant

Comment (alexbrdn) [reply libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java]: maybe even the whole pattern

File: libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java (hunk 1 above)
Comment (maierthomas) [reply libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java]: corrected

**Comment by maierthomas** [comment]
- renamed `validateLicense` to `validateNewLicense`
- removed if condition (isSet) for `license.shortname`
- removed assertId and pattern check for `license.id`

**Pull request merged**